### ISSUE TITLE
Install problem

### ISSUE THREAD

**Issue by mind-mind** [cri-o/cri-o#4692]
I can't install CRI-O on Debian 10. I follow this https://cri-o.io 

it doesn't work.

**Comment by haircommander** [comment]
what version of cri-o and for what arch?

**Comment by mind-mind** [comment]
What do you mean by arch?

I use this command

cat <<EOF | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list 
deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/Debian_Unstable/ /
EOF


cat <<EOF | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable:1.20:1.20.0.list
deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/cri-o:1.20:1.20.0/Debian_Unstable/ /
EOF


curl -L https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.20:1.20.0/Debian_Unstable/Release.key | sudo apt-key --keyring /etc/apt/trusted.gpg.d/libcontainers.gpg add -


curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/Debian_Unstable/Release.key | sudo apt-key --keyring /etc/apt/trusted.gpg.d/libcontainers.gpg add -

sudo apt-get update
sudo apt-get install cri-o cri-o-runc

Cr: https://kubernetes.io/docs/setup/production-environment/container-runtimes/ 

it was unknown cri-o

**Comment by haircommander** [comment]
what is the actual error?

**Comment by mind-mind** [comment]
I follow the step add repository.

First, it wasn’t found the repository.

Second, when add key -, I got ok, but when I use sudo apt install cri-o.

 It was still unknown.

**Comment by haircommander** [comment]
Are you able to paste the output?

**Comment by mind-mind** [comment]
Sorry, I did in front of my mac, I turn on it now.
After I use the command as I explain above here is the output.

Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package cri-o

**Comment by mind-mind** [comment]
it seemed like the repository doesn't work.

**Comment by haircommander** [comment]
that's really weird. I spun up a debian 10 container, and did run into issues, but not that issue...

are you able to install 1.19?

**Comment by mind-mind** [comment]
Umm, could you please give me a repository? 

For example, 

curl -L https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:$VERSION/$OS/Release.key | apt-key add -
curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/Release.key | apt-key add -

Since I'm not sure should I replease some value on $OS or $VERSION.

As I follow this https://cri-o.io   

I also got the error as shown in picture as well. 
The key seems fine.  Not sure why it not found.


<img width="945" alt="Screen Shot 2021-03-26 at 3 13 38 AM" src="https://user-images.githubusercontent.com/68138683/112537678-90b6f700-8de1-11eb-9358-29191e153425.png">
<img width="1134" alt="Screen Shot 2021-03-26 at 3 14 22 AM" src="https://user-images.githubusercontent.com/68138683/112537689-93b1e780-8de1-11eb-9221-de7746b2cc68.png">

**Comment by mind-mind** [comment]
Dear @haircommander 

I follow this https://insujang.github.io/2019-11-21/installing-kubernetes-and-crio-in-debian/ 

It was work, but it would be better if there are any updates about the version from this source is 1.15.

As I thought, seems like the repository is wrong.
I will close this.

**Comment by haircommander** [comment]
we don't support that PPA anymore, the only supported source is OBS, I'm afraid. I am sorry for the delay and the issues

**Comment by mind-mind** [comment]
Well, I can install and it works for some reason.
 
But about OBS on k8s document and https://cri-o.io today
both don't work even use another OS. I use ubuntu once, but it cannot found any repository.

Could you please give some source to searching or something? 
I'd very much appreciate that. Thank you.

**Comment by haircommander** [comment]
I wonder if there's a dns issue. to verify, have you tried these instructions *exactly* https://github.com/cri-o/cri-o/blob/master/install.md#apt-based-operating-systems ?

**Comment by mind-mind** [comment]
I think I did, but I'll try again.

**Comment by mind-mind** [comment]
<img width="1286" alt="Screen Shot 2021-03-26 at 9 47 46 PM" src="https://user-images.githubusercontent.com/68138683/112649460-0b812000-8e7d-11eb-88c5-51303fc455f1.png">

As you can see, I think there is a problem with the repository or I key something wrong?

**Comment by mind-mind** [comment]
<img width="1286" alt="Screen Shot 2021-03-26 at 9 47 46 PM" src="https://user-images.githubusercontent.com/68138683/112649697-3bc8be80-8e7d-11eb-9134-0b9ec15d0794.png">

**Comment by haircommander** [comment]
can you try running `apt-get install ca-certificates`

**Comment by haircommander** [comment]
ref: https://stackoverflow.com/questions/21338721/gpg-no-valid-openpgp-data-found

alternatively, can you do the curl command without apt-key?

**Comment by haircommander** [comment]
also, sanity check, you have `gnupg` installed right?

**Comment by mind-mind** [comment]
Yeah, because 

<img width="1113" alt="Screen Shot 2021-03-26 at 9 56 13 PM" src="https://user-images.githubusercontent.com/68138683/112651118-aaf2e280-8e7e-11eb-96d1-12ff27931f05.png">

version 1.17 seems fine and apt-key is ok, but anyway it was not found

<img width="1182" alt="Screen Shot 2021-03-26 at 9 56 38 PM" src="https://user-images.githubusercontent.com/68138683/112651201-c100a300-8e7e-11eb-9a9b-2a458ab7d23f.png">

**Comment by haircommander** [comment]
okay I also ran into that issue, so now we're in the same place. This is a legitimate issue with the package. In the meantime, you could follow the instructions to allow the insecure sources, or give me some  time to figure out why the package is weird

**Comment by haircommander** [comment]
fwiw, I believe `1.19` does not have this issue (not to be confused with `1.19:1.19.0`

**Comment by mind-mind** [comment]
about 1.19, I have problem with that.

<img width="1180" alt="Screen Shot 2021-03-26 at 10 05 15 PM" src="https://user-images.githubusercontent.com/68138683/112651959-7cc1d280-8e7f-11eb-861d-25d944d09e9b.png">

**Comment by mind-mind** [comment]
<img width="1180" alt="Screen Shot 2021-03-26 at 10 10 01 PM" src="https://user-images.githubusercontent.com/68138683/112652556-1be6ca00-8e80-11eb-820a-d42bb9f1d7cb.png">

1.19:1.19.1 look ok, but

<img width="1245" alt="image" src="https://user-images.githubusercontent.com/68138683/112652585-23a66e80-8e80-11eb-8698-474c8f6a5f43.png">

The same problem with 1.17

**Comment by haircommander** [comment]
weird. are you able to download `1.19.1`?

**Comment by mind-mind** [comment]
No, it was as you can see in the picture above.

apt-key fine with 1.19.1, but not found cri-o. 
It is the same as the issue with 1.17

**Comment by mind-mind** [comment]
While waiting for you to get back, I tried with crio 1.15, but it can not starts the pod.

Installed by PPA

Use k8s init and join with another was fine, but when trying to start pod,

got error like 

Failed to create pod sandbox: rpc error: code = Unknown desc = container create failed

Seem likes it cannot connect with flannel even though I already install flannel.
So, the pod doesn't ready cause cannot pull any images.

The status of crio was an error as well.

I'm not sure because of version crio whether or not, but it should be works.

**Comment by mind-mind** [comment]
<img width="1242" alt="image" src="https://user-images.githubusercontent.com/68138683/112715069-be4e8e00-8f10-11eb-9bcc-08beb5bf2ae3.png">

#4183

I found the reason now. It may not update I guess.

**Comment by mind-mind** [comment]
Hi @haircommander 

I just come to tell you that. I indicated the problem. Your repository is the problem. I tried to use another OS with multiple versions. There are no repository works. 

I need you to do something with your repository.
I need to use CRIO. Also, I tried to clone Github to install in a source, but there is an error about go as well.

**Comment by haircommander** [comment]
yeah sorry I'll try to take a look today

**Comment by mind-mind** [comment]
Don't be. Just I'd appreciate you doing that.

**Comment by mind-mind** [comment]
Hi, I found the way to install from source GitHub, but I got this error.

build github.com/cri-o/cri-o/cmd/crio: cannot find module for path archive/tar

Actually, I success to install once, but when start cri-o there is an error, so I re-install again and got this error

**Comment by mind-mind** [comment]
Well, I fixed everthing. Thanks anyway.

**Comment by haircommander** [comment]
thanks for reporting the issue and figuring it out. I will leave this open to address the debian issues

**Comment by mind-mind** [comment]
I think I know about this problem a little bit. Actually, Debian10 doesn't support many repository, but I don't know why either.

**Comment by mind-mind** [comment]
Hi, haircommandaer,

It's been for a while.

I tried CRI-O with Raspbian, repository cannot use either and also I can't install CRI-O from the source with Raspbian because Raspbian doesn't has go version 12 plus.

I use Raspbian armv7l 32 bit

**Comment by lsm5** [comment]
@haircommander i'll give this a try and report back.

**Comment by lsm5** [comment]
@haircommander @mind-mind RE: kubic repo `does not have a Release file`, I'd suggest cleaning out prior repo additions along with ensuring any kubic related files in `/etc/apt/trusted.gpg.d` are also deleted and then retrying the installation steps

**Comment by lsm5** [comment]
on debian 10, you may quite likely hit a libseccomp2 dependency issue which can be resolved with:

```
sudo apt install libseccomp2 -y -t buster-backports
```

**Comment by lsm5** [comment]
> @haircommander @mind-mind RE: kubic repo `does not have a Release file`, I'd suggest cleaning out prior repo additions along with ensuring any kubic related files in `/etc/apt/trusted.gpg.d` are also deleted and then retrying the installation steps


fwiw, I tried the install steps with cri-o 1.19 kubic repo on a debian 10 env, and it went just fine for me.

**Comment by mind-mind** [comment]
@lsm5  Thank you. I'II try repo Debian again. 

Anyway, I still want to know about Rasbian.

Thank you so much.

**Comment by lsm5** [comment]
> @lsm5 Thank you. I'II try repo Debian again.
> 
> Anyway, I still want to know about Rasbian.

It should be the exact same as Debian instructions with the exception of `s/Debian_10/Raspbian_10/g` in the repository URLs. Let me know if that doesn't work.

**Comment by haircommander** [comment]
here's a PR adding instructions for libseccomp bump based on @lsm5 's recommendation: https://github.com/cri-o/cri-o/pull/4854

**Comment by mind-mind** [comment]
Okay.

I tried clean repo from /etc/apt/trusted.gpg.d 
and follow the step again.  I remove all of this.

I'm not sure about version 1.19. Did you mean 1.19:1.19.1, 
1.19:1.19.2 or 1.19:1.19.3

I'm not sure that I mean some stop or something 

And here what I got:

root@master:/home/admin# ls /etc/apt/trusted.gpg.d 
root@master:/home/admin# VERSION=1.19:1.19.1
root@master:/home/admin# OS=Debian_10
root@master:/home/admin# echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
root@master:/home/admin# echo "deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/ /" > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable:cri-o:$VERSION.list
root@master:/home/admin# curl -L https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:$VERSION/$OS/Release.key | apt-key add -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   416  100   416    0     0    517      0 --:--:-- --:--:-- --:--:--   518
100   417  100   417    0     0    425      0 --:--:-- --:--:-- --:--:--     0
100   418  100   418    0     0    361      0  0:00:01  0:00:01 --:--:--   361
100   419  100   419    0     0    314      0  0:00:01  0:00:01 --:--:--   314
100   420  100   420    0     0    278      0  0:00:01  0:00:01 --:--:--   278
100   421  100   421    0     0    250      0  0:00:01  0:00:01 --:--:--   250
100  1093  100  1093    0     0    587      0  0:00:01  0:00:01 --:--:--   587
OK
root@master:/home/admin# curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/Release.key | apt-key add -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1093  100  1093    0     0   2074      0 --:--:-- --:--:-- --:--:--  2074
OK
root@master:/home/admin# apt-get update
Hit:1 http://security.debian.org/debian-security buster/updates InRelease
Hit:2 http://cdn-aws.deb.debian.org/debian buster InRelease                                                       
Hit:3 http://cdn-aws.deb.debian.org/debian buster-updates InRelease                                               
Err:1 http://security.debian.org/debian-security buster/updates InRelease                                         
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY AA8E81B4331F7F50 NO_PUBKEY 112695A0E562B32A
Err:2 http://cdn-aws.deb.debian.org/debian buster InRelease                                                       
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138 NO_PUBKEY DCC9EFBF77E11517
Err:3 http://cdn-aws.deb.debian.org/debian buster-updates InRelease                                               
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138
Get:4 https://packages.cloud.google.com/apt kubernetes-xenial InRelease [9383 B]                     
Ign:5 http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10  InRelease
Hit:6 http://cdn-aws.deb.debian.org/debian buster-backports InRelease                                
Err:7 http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10  Release
  404  Not Found [IP: 195.135.221.134 80]
Err:6 http://cdn-aws.deb.debian.org/debian buster-backports InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138
Hit:8 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/Debian_10  InRelease
Reading package lists... Done
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://security.debian.org/debian-security buster/updates InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY AA8E81B4331F7F50 NO_PUBKEY 112695A0E562B32A
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://cdn-aws.deb.debian.org/debian buster InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138 NO_PUBKEY DCC9EFBF77E11517
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://cdn-aws.deb.debian.org/debian buster-updates InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138
E: The repository 'http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10  Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://cdn-aws.deb.debian.org/debian buster-backports InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138
root@master:/home/admin# apt-get install cri-o cri-o-runc
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package cri-o

**Comment by haircommander** [comment]
unfortunately, removing everything in `/etc/apt/trusted.gpg.d` gave me the same errors. I was able to fix by finding every instance of `NO_PUBKEY $hash` and running
```shell
apt-key adv --keyserver keyserver.ubuntu.com --recv-keys $hash
```
Then, my machine reproduced the
```
E: The repository 'http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10  Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
```
issue, but I was able to work around it by running:
```shell
# echo "deb [trusted=yes] https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
echo "deb [trusted=yes] http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/ /" > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable:cri-o:$VERSION.list
```

(note the added [trusted=yes])
and then running apt-update && apt install

**Comment by lsm5** [comment]
@haircommander hmm, that's news to me, but great that you resolved it!

**Comment by mind-mind** [comment]
Unfortunately, I hope I can fix it too. 
I did everything that you recommend but still doesn't work. 
I use AWS EC2, I'm not sure what platform that you use, but if you use AWS too. I think it should be works. 

Umm, I never succeeded with install by repository, but I can install by source. 

But, it cannot do that with Raspbian. Because make require go version 12+.
Raspbian armv7l as far as I know it doesn't have a new version than go version 11.

root@master:~# apt-key adv --keyserver keyserver.ubuntu.com --recv-keys $hash
Executing: /tmp/apt-key-gpghome.YRWRlPZKy8/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys
root@master:~# echo "deb [trusted=yes] https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
root@master:~# echo "deb [trusted=yes] 

root@master:~# apt update
..........................................


Err:8 http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10  Packages
  404  Not Found [IP: 195.135.221.134 80]
Ign:10 http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10  Translation-en
Reading package lists... Done
E: Failed to fetch http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10/Packages  404  Not Found [IP: 195.135.221.134 80]
E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by lsm5** [comment]
> Unfortunately, I hope I can fix it too.
> I did everything that you recommend but still doesn't work.
> I use AWS EC2, I'm not sure what platform that you use, but if you use AWS too. I think it should be works.
> 
> Umm, I never succeeded with install by repository, but I can install by source.
> 
> But, it cannot do that with Raspbian. Because make require go version 12+.
> Raspbian armv7l as far as I know it doesn't have a new version than go version 11.
> 
> root@master:~# apt-key adv --keyserver keyserver.ubuntu.com --recv-keys $hash Executing: /tmp/apt-key-gpghome.YRWRlPZKy8/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys root@master:~# echo "deb [trusted=yes] https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
> root@master:~# echo "deb [trusted=yes]
> 
> root@master:~# apt update
> ..........................................
> 
> Err:8 http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10 Packages
> 404 Not Found [IP: 195.135.221.134 80]
> Ign:10 http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10 Translation-en
> Reading package lists... Done
> E: Failed to fetch http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.19:1.19.1/Debian_10/Packages 404 Not Found [IP: 195.135.221.134 80]
> E: Some index files failed to download. They have been ignored, or old ones used instead.

It looks like you're trying to setup Debian_10 repo on raspbian. I would suggest getting the Raspbian_10 repo. It should be just a matter of replacing `Debian` with `Raspbian` everywhere in the repo urls.

**Comment by mind-mind** [comment]
Well, I don't think so. I turn raspberry pi off before tried this, but I'II tries to make it work and back to you guys later.

**Comment by mind-mind** [comment]
Hello guys,

I can install it already the thing is I have to fix version=1.19 instead of 1.19:1.19.1 and it works.

I think I can close this right?

**Comment by haircommander** [comment]
glad the installation finally worked! thanks for your patience!
